enh: move from yarn to npm and update docs (#841)
all-contributors/all-contributors: docs/project/contribute.md

## Running the site locally:

> All Contributors currently uses [Docusaurus 1.x](https://docusaurus.io) for our documentation website.

Author: I added instructions about using npm, and running a linter locally.

To build the docs locally:

* <https://tutorial.docusaurus.io/docs/intro>

1. Make sure `npm` is installed on your machine. Use the [nodejs documentation](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) if you aren't sure how to install `npm`.

To check that npm is installed run the following in your favorite shell:

`npm -v`

> Next, make sure you have atleast node version 0.18 or higher. To check the version,

Reviewer: ```suggestion
Next, make sure you have at least node version 0.22 (LTS) or higher. To check the version,
```

I would recommend the current LTS version. 0.18 is no longer maintained and is EOL.

Author: Good suggestion! I am building locally using node 22. I'm VERY WORRIED about upgrading too much else until we bump to a newer version of Docusaurus!!

use:

`node -v`

> If you don't have atleast version `.18` or higher, please install node or upgrade your current version.

Reviewer: ```suggestion
If you don't have at least version `0.22` or higher, please install Node or upgrade your current version.
```

> Install the newest (deprecated) version of Docusaurus:

Reviewer: ```suggestion
Install docusaurus v1 (1.14.7). *Note: This project does not yet support docusaurus v2 or v3; though, we hope to do so in the future.*
```

> `npm install docusaurus@1.14.7 --save-dev`

Reviewer 2: Maybe we should add docusaurus to the dependencies so we can just run `npm install`?

Author: So I just looked - it's there - i'll update it and you are correct, we can then just use npm install. Thanks for catching that!

Then install the dependencies:

`npm install`

Once you have completed the above, you can launch a server locally that will build and run the docs locally. In your favorite shell run:

`npm run start`

* Then, go to `http://localhost:3000` and you will see the all contributors website there!

> * Refresh the page as you make changes to the markdown pages in site locally

Reviewer: ```suggestion
* Refresh the browser page as needed when editing markdown pages.
```

### Make changes - linting your contribution

If you make changes to a markdown file in the site, then our linting action will
check those changes. To run the linter yourself on a file locally:

First install markdownlint-cli:

`npm install -g markdownlint-cli`

Then run markdownlint on the file (or files) that you wish to check:

`markdownlint --fix "docs/project/contribute.md"`

Or you can run it on all files in the docs directory, like this:

> markdownlint --fix "docs/**.md"

Reviewer: ```suggestion
`markdownlint --fix "docs/**.md"`
```
